cloud/pkg/cloudstream/streamserver.go

```diff
@@ -228,7 +228,23 @@ func (s *StreamServer) getExec(request *restful.Request, response *restful.Respo
 		}
 	}()
 
-	sessionKey := strings.Split(request.Request.Host, ":")[0]
+	// extract pod namespace and pod name from request
+	meta := strings.Split(request.Request.URL.Path, "/")
+	namespaceName := meta[2]
+	podName := meta[3]
+
+	kubeClient := client.GetKubeClient()
+	if kubeClient == nil {
+		klog.Exitf("cannot get kube client\n")
+		return
+	}
+
+	pod, err := kubeClient.CoreV1().Pods(namespaceName).Get(context.Background(), podName, v1.GetOptions{})
+	if err != nil {
+		klog.Errorf("get pod %s/%s failed: %v\n", namespaceName, podName, err)
+		return
+	}
+	sessionKey := pod.Spec.NodeName
```

Comment:
Looks good, thanks for the fixing, can we wrap a function to get the sessionKey? So we can reuse it in `logs/exec/metrics`.

Reply:
done. But only logs/exec is modified. I do not have a good idea for metrics. Most of metrics url doesn't contain node name:

![image](https://user-images.githubusercontent.com/6214067/147546327-0e789037-862b-4853-b5c4-ae25fdd2ca27.png)